.github/workflows/ci.yml

```diff
@@ -0,0 +1,149 @@
+name: asn1-codec
+
+on: [pull_request, push]
+    # Trigger this workflow on code pushes and pull requests
+
+jobs:
+  asn1-codec:
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:jammy-20230126
+      env:
+        working-directory: /asn1_codec
+        BUILD_WRAPPER_OUT_DIR: "$GITHUB_WORKSPACE/bw-output"
+      options: "--user root"
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+        # Checkout the code from the repository
+
+      - name: Install necessary dependencies
+        run: |
+          apt update
+          apt-get -y install sudo wget curl gnupg lsb-release gcovr unzip gcc-multilib libasan*
+          sudo apt-get -y install software-properties-common
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+          sudo add-apt-repository -y "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+          sudo curl -L "https://github.com/docker/compose/releases/download/1.24.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          chmod +x /usr/local/bin/docker-compose
+          sudo apt-get -y update
+          sudo apt-get -y install docker-ce
+        working-directory: ${{ env.working-directory }}
+        
+      - name: Install g++ and related dependencies
+        run: |
+          sudo apt-get -y install build-essential
+          sudo apt -y install cmake g++ libprotobuf-dev protobuf-compiler
+          sudo apt -y install apt-transport-https ca-certificates curl software-properties-common
+          export CXX="g++"
+          sudo apt install -y gcc-12 g++-12
+          sudo apt -y install libpng-dev libgraphicsmagick1-dev libcairo2-dev
+        working-directory: ${{ env.working-directory }}
+
+      - name: Install librdkafka
+        run: |
+          sudo apt-get -y update && sudo apt-get -y upgrade
+          sudo apt-get -y install automake flex bison
+          sudo apt-get -y install build-essential
+          sudo apt -y install cmake g++ libprotobuf-dev protobuf-compiler
+          sudo apt-get install autotools-dev
+          sudo apt -y install apt-transport-https ca-certificates curl software-properties-common
+          git clone --depth 1 https://github.com/confluentinc/librdkafka.git librdkafka
+          cd librdkafka
+          cmake -H. -B_cmake_build
+          cmake --build _cmake_build
+          cmake --build _cmake_build --target install
+        working-directory: ${{ env.working-directory }}

```

Comment:
All references to apt will need to be replaced with apk-equivalent statements when the incoming CDOT alpine-image changes are merged into USDOT develop.